text/0000-add-pack-interact.md

```diff
@@ -0,0 +1,83 @@
+# Meta
+[meta]: #meta
+- Name: Add Pack Interact
+- Start Date: 2020-04-05
+- Author(s): aemengo
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Pack should provide an interactive mode for building images, updating dependencies, and other user workflows. Please see [mockup](https://storage.googleapis.com/artifacts.cf-garden-windows-dev.appspot.com/pack-interact-edit.mp4) for concept.
+
+# Definitions
+[definitions]: #definitions
+
+* [Pack](https://github.com/buildpacks/pack) - A tool within the buildpacks ecosystem for converting code into runnable images, in a local environment.
+* Build plan - A sequence of implementation buildpacks prescribed by the detect phase.
+
+# Motivation
+[motivation]: #motivation
+
+The **casual application developer** has trouble seeing the value proposition of buildpacks. They also encounter a high learning curve when using our tooling. We've seen evidence of this through user interviews and other channels. By providing a more interactive experience, the casual application developer will be more enticed to remain within the buildpacks ecosystem. Pack will act as the gateway tool to more advanced parts of the ecosystem.
+
+# What it is
+[what-it-is]: #what-it-is
+
+When a casual application developer runs `pack interact` on the command-line in the working directory of their application, they are provided with an interactive panel where they can perform activities related to converting code into runnable images. The following **2 user workflows** will be introduced in the interactive panel:
+
+1. **Break between detection and compilation.** Buildpacks should be known as a tool that writes (90%) of your Dockerfiles for you. You should be allowed to **see and intervene** with the "build plan" autogenerated for you. And when you **(b) build**, you see the **implementation buildpacks** (as steps) being traversed. It is important that a little description is provided for each implementation buildpack to provide insight into what is happening.
+1. **Updates.** One thing Dockerfiles can't do is update hardcoded dependencies. We should be able to show when buildpacks, lifecycle, stack, etc, can be upgraded so that the user gets this sense of value.
+
+---
+
+This RFC acts as a gateway to subsequent functionality. Follow-up RFCs could include:
+
+1. **CVE detection.** A user should be able to view CVEs that a **(r) rebase** will remove.
+1. **Build Hijack.** A user should be able to open a shell for debugging, into a failed build container.
+1. **Edit Build Plan.** A user should be able to modify a build plan, before building.  
```

Comment:
Is this the same as the "**Break between detection and compilation.** or something different?

Reply:
I'm thinking that it's something different.

* **Break between detection and compilation** just allows you to see and continue with the build plan.
* **Edit Build Plan** is something enabled with **(e) edit**. You perform it during the _break_ above. Since I don't know what it looks like yet, I think it should be reserved for a separate RFC.